subsys/net/lib/lwm2m_client_utils/lwm2m/gnss_assistance_obj.c

```diff
@@ -0,0 +1,319 @@
+				  uint32_t bytes_downloaded)
+{
+	int err;
+
+	if (assist_buf == NULL) {
+		return -ENOMEM;
+	}
+
+	memcpy(&assist_buf[bytes_downloaded], &data[0], data_len);
+
+	if (last_block) {
+		err = nrf_cloud_agps_process(assist_buf, bytes_downloaded);
+		if (err) {
+			LOG_WRN("Unable to process A-GPS data, error: %d", err);
+		} else {
+			LOG_INF("A-GPS data processed");
+		}
+		k_free(assist_buf);
+		assist_buf = NULL;
+		bytes_downloaded = 0;
+		return err;
+	}
+
+	return 0;
+}
+#endif
+
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_PGPS)
+static int gnss_assist_write_pgps(uint8_t *data, uint16_t data_len, bool last_block,
+				  uint32_t bytes_downloaded)
+{
+	int err;
+
+	if (bytes_downloaded == 0) {
+		err = nrf_cloud_pgps_begin_update();
+	}
+	err = nrf_cloud_pgps_process_update(data, data_len);
+	if (last_block) {
+		if (err) {
+			LOG_WRN("Unable to process P-GPS data, error: %d", err);
+		} else {
+			LOG_INF("P-GPS data processed");
+		}
+		err = nrf_cloud_pgps_finish_update();
+	}
+	return err;
+}
+#endif
+
+static int gnss_assist_write_cb(uint16_t obj_inst_id, uint16_t res_id,
+			    uint16_t res_inst_id, uint8_t *data, uint16_t data_len,
+			    bool last_block, size_t total_size)
+{
+	int err = 0;
+	static uint32_t bytes_downloaded;
+
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS)
+	if (assist_type == ASSISTANCE_REQUEST_TYPE_AGPS) {
+		err = gnss_assist_write_agps(data, data_len, last_block);
+	}
+#endif
+
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_PGPS)
+	if (assist_type == ASSISTANCE_REQUEST_TYPE_PGPS) {
+		err = gnss_assist_write_pgps(data, data_len, last_block, bytes_downloaded);
+	}
+#endif
+
+	if (err) {
+		LOG_ERR("Error writing assistance data (%d)", err);
+		return err;
+	}
+
+	bytes_downloaded += data_len;
+
+	return 0;
+}
+
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS)
+int location_assist_agps_alloc_buf(size_t buf_size)
+{
+	/* Free the memory if it wasn't unallocated in the last final package */
+	if (assist_buf != NULL) {
+		k_free(assist_buf);
+		assist_buf = NULL;
+	}
+
+	assist_buf = k_malloc(buf_size);
+
+	if (assist_buf == NULL) {
+		return -ENOMEM;
+	}
+
+	return 0;
+}
+
+void location_assist_agps_request_set(uint32_t request_mask)
+{
+	LOG_INF("Requesting A-GPS data, mask 0x%08x", request_mask);
+	/* Store mask to object resource */
+	agps_mask = request_mask;
+	assist_type = ASSISTANCE_REQUEST_TYPE_AGPS;
+}
+#endif /* CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS*/
+
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS_FILTERED)
+void location_assist_agps_set_elevation_mask(int32_t elevation_mask)
+{
+	satellite_elevation_mask = elevation_mask;
+}
+#endif /* CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_AGPS_FILTERED */
+
+#if defined(CONFIG_LWM2M_CLIENT_UTILS_LOCATION_ASSIST_PGPS)
+int location_assist_pgps_set_prediction_count(int32_t predictions)
+{
+	if (predictions < 1 || predictions > 168) {
```

Comment:
Magic values. I would define those. There seem to be many on the functions which follow this one.

Reply:
Done